File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/Conne/ProductForm.ts
import { StringEditor, DecimalEditor, ServiceLookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ProductForm {
    Name: StringEditor;
    Description: StringEditor;
    Price: DecimalEditor;
    Catid: ServiceLookupEditor;
    Photo: StringEditor;
}

export class ProductForm extends PrefixedContext {
    static readonly formKey = 'Conne.Product';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProductForm.init)  {
            ProductForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = ServiceLookupEditor;

            initFormType(ProductForm, [
                'Name', w0,
                'Description', w0,
                'Price', w1,
                'Catid', w2,
                'Photo', w0
            ]);
        }
    }
}

File: AdminPanel/AdminPanel.Web/Modules/Conne/Product/ProductForm.cs
using Serenity.ComponentModel;

namespace AdminPanel.Conne.Forms;

[FormScript("Conne.Product")]
[BasedOnRow(typeof(ProductRow), CheckNames = true)]
public class ProductForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public int Catid { get; set; }
    public string Photo { get; set; }
}